Cuneiform transliteration

EricJHarvey opened the pull request:
This table differs from the existing table `cuneiform-transliterated.utb`. That table seems to adopt UEB dot patterns for the various diacritic marks, whereas this table uses a code developed specifically for cuneiform and other ancient languages.

I've called this one the "Academic" variant to differentiate it from the one that uses UEB dot patterns.

EricJHarvey commented:
This table differs from the existing table cuneiform-transliterated.utb. That table seems to adopt UEB dot patterns for the various diacritic marks, whereas this table uses a code developed specifically for cuneiform and other ancient languages. 
I've called this one the "Academic" variant to differentiate it from the one that uses UEB dot patterns.

egli commented:
It seems that there is a problem when testing `braille-specs/cuneiform-transliterated.yaml`. That file seems to pick `cuneiform-transliteration-academic.utb` over `cuneiform-transliterated.utb`. I guess the metadata needs some refinement or the query in `braille-specs/cuneiform-transliterated.yaml`.

bertfrees commented:
@EricJHarvey This looks great. I'm only missing some metadata. I would like to know which braille code this table implements: is there some more formal documentation? You included some links in the YAML file, but it is not clear enough what is behind those those. At this point, for me, there is not a strong enough case yet for having two tables that seemingly serve the same purpose.

I also have some more minor comments. See my comments in the code.

bertfrees commented on tables/cuneiform-transliteration-academic.utb:
```diff
@@ -0,0 +1,713 @@
+#-index-name: Cuneiform (Academic)
+#-display-name: Cuneiform Transliteration and Normalization,  Academic
+#
+#+language: akk-Latn
+#+language: xhu-Latn
+#+language: uga-Latn
+#+language: peo-Latn
+#+language: sux-Latn
+#+language: xeb-Latn
+#+language: elx-Latn
+#+language: hit-Latn
+#+language: xur-Latn
+#+language: xlu-Latn
```
I want to make sure that this table actually target all of these languages. In addition, are there any languages missing from this list? In the description below you mention Hebrew, Aramaic, Coptic, ... I don't see these here.

bertfrees commented on tables/cuneiform-transliteration-academic.utb:
```diff
@@ -0,0 +1,713 @@
+#+type: literary
+#+contraction: no
+#+dots: 6
+#+unicode-range: ucs4
```
We need to check whether this table has Unicode characters above U+FFFF. The other table has U+D1D30, and perhaps you have just copied the metadata section from the other table.

EricJHarvey commented:
Yes, of course. I will try to address the main issues raised below:
The system in cuneiform-translitaeration-academic.utb was created in 2017 and has been implemented in Duxbury since 2018. You can find their documentation of it here:
https://www.duxburysystems.com/documentation/dbt12.5/languages/semitic_tbl.htm
Both Semitic” and “Cuneiform” are slight misnomers, since the code handles languages like Hittite and Persian, which are not Semitic, and like Aramaic and Hebrew, which are not cuneiform. 

I originally intended to use cuneiform-transliterated.utb and suggest refinements if necessary, but found it uses a completely different system. The diacritics are crucial to transliteration, and they are different between cuneiform-transliterated.utb (UEB) and the system I was familiar with. The differences meant a new table was needed, and it also meant I couldn’t “include en-ueb-chardefs.uti” since the two contain many conflicts.

The following examples show the differences in output from the two tables:
Input 1:
% echo "du-ul-lu ú-šá-az-ba-lu i-gi4-gi4" | lou_translate -f unicode.dis,cuneiform-transliterated.utb          
⠙⠥⠤⠥⠇⠤⠇⠥⠀⠘⠌⠥⠤⠩⠘⠌⠁⠤⠁⠵⠤⠃⠁⠤⠇⠥⠀⠊⠤⠛⠊⠙⠤⠛⠊⠙
% echo "du-ul-lu ú-šá-az-ba-lu i-gi4-gi4" | lou_translate -f unicode.dis,cuneiform-transliteration-academic.utb
⠙⠥⠤⠥⠇⠤⠇⠥⠀⠆⠥⠤⠩⠆⠁⠤⠁⠵⠤⠃⠁⠤⠇⠥⠀⠊⠤⠛⠊⠼⠙⠤⠛⠊⠼⠙

Input 2:
% echo "ša₂ gišBANŠUR 5 šap-pi-MEŠ KU₃.SI₂₂" | lou_translate -f unicode.dis,cuneiform-transliterated.utb
⠩⠁⠢⠼⠃⠀⠛⠊⠩⠃⠁⠝⠠⠩⠥⠗⠀⠑⠀⠩⠁⠏⠤⠏⠊⠤⠍⠑⠠⠩⠀⠅⠥⠢⠼⠉⠲⠎⠊⠢⠼⠃⠢⠼⠃
% echo "ša₂ gišBANŠUR 5 šap-pi-MEŠ KU₃.SI₂₂" | lou_translate -f unicode.dis,cuneiform-transliteration-academic.utb
⠩⠁⠼⠃⠀⠛⠊⠩⠨⠠⠃⠁⠝⠩⠥⠗⠀⠼⠑⠀⠩⠁⠏⠤⠏⠊⠤⠨⠠⠍⠑⠩⠀⠨⠠⠅⠥⠼⠉⠲⠨⠠⠎⠊⠼⠃⠃

I added many of the punctuation rules from en-ueb-chardefs.uti to my table, since they do not conflict. I will happily add credit and a copyright line, but I don’t think I can “include” the whole table.
Another, more laborious, option would be to split en-ueb-chardefs.uti into two .uti tables—one for diacritics and a second one for other character definitions. This would be a hassle, but might lead to fewer conflicts down the road.

EricJHarvey commented:
@bertfrees Is there a list of language codes so i can double-check and add Hebrew,etc.?

bertfrees commented:
Thanks for that link to the Duxbury documentation. It seems they have pretty good documentation! Feel free to include that in the table header (in absence of more formal and complete documentation). Alternatively, you may create a similar web page in our https://github.com/liblouis/braille-specs repo.

For language codes, I usually consult Wikipedia, or there is also this list that you can search in: https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry.

> The differences meant a new table was needed

OK. But my problem is that it is currently not clear enough what the differences are / why the differences are there, and I'd rather not have that ambiguity in Liblouis. Perhaps Matt's table is wrong, which means your table could replace it. Or they are actually two different tables used in different contexts, but then the user must have more information to pick one.

> and it also meant I couldn’t “include en-ueb-chardefs.uti” since the two contain many conflicts.

In some cases it may *really* not be possible to include another table, but I would say in most cases something can be done. We have to deal with this issue all the time. There are often conflicts, but by including the table at the end of your table, the rules defined in your table override the rules in the included table. Splitting en-ueb-chardefs.uti might indeed also be appropriate. We're happy to help with this refactoring work, and James (from the UEB table) might be too.

EricJHarvey commented:
Thank you, @bertfrees, for the help. I checked the language listings and removed Luwian and Old Persian, as well as adding a few others that will be handled well. If the list looks large, it's because many of those languages use the same alphabet and share transliteration practices.
I also removed the "ux4" range in the metadata because I removed the characters in that unicode range. If we find them in real-world transliteration we can add them back in.
I also added information about the code and about print conventions for transliteration of the various languages.

As for whether we need one table or two, I get what you mean now. I only know of the one braille code for transliteration and only contributed to one. It seems from the metadata that Matt's table was also meant to represent this code, so perhaps this table can be a replacement rather than a second table.

bertfrees commented:
Thanks @EricJHarvey for your latest version. That solves most of my questions!

@yeatersink Please have a look at this proposed new table and accompanying test data from Eric. At first sight, it has the same purpose as the existing table for transliterated Cuneiform that you provided last year.

Eric says he only knows of one braille code for transliterated Cuneiform, and only contributed to one. It is the same as the [Duxbury table called "Semitic"](https://www.duxburysystems.com/documentation/dbt12.5/languages/semitic_tbl.htm), which you also contributed to. The table is quite substantiated with links to online documentation.

Both of us are wondering why your Liblouis table adopts UEB dot patterns for the various diacritic marks. Eric's table and the Duxbury table it's based on use a code developed specifically for the diacritic marks. Does your table implement a different code? It doesn't seem to be just mistakes in the table. Your tests (cuneiform-transliterated.yaml) confirm that your table behaves the way it does, so I guess it must be deliberate?

yeatersink commented:
Hi @bertfrees  I am glad that Eric is contributing.  The Duxbury project was initiated by me in 2013 and I was the one who really developed the braille code. Eric did come to the committee later and his input is valuable. There may be a few differences because I was actually developing the code while in class. lol . But as far as I know, everyone was on board.  

The braille code at DBT was proposed and in testing, and the last conversations I had with David Holladay was updating the diacritics  to follow UEB for consistency.  Everything is following UEB these days. If David Holladay did not update that before he passed, I don't know what to say about that. It Sounds like we may need to update the table with DBT as well. 

The facts are that David was literally working while he was on Hospice. I am so thankful for all he did, and i want every thing I do to honor his legacy. I hope that this does not come across   strong. We literally would not be where we are at without his work.  

The way that the Cuneiform  table is set up, it doesn't seem to make sense for things not to follow the UEB standard. Especially since so many seem to be updating their tables so that they do too.   

 This table is pretty solid. 

The table that really needs work is the German berger . The names and signs are  correct, but not preferred.

bertfrees commented:
Thanks for the explanation! It makes more sense now.

yeatersink commented:
I would like to make a couple of points. 1 We need to follow our braille authority standards. There is nothing Academic about having a different braille for diacritics. 
2 If you do decide to have another Transliteration table, please do not call this an Academic version because the table that would more align with  academic standards would be the First Transliteration table that i committed last year.  

When a student engages with the table, they will expect the diacritics  as are  represented by UEB.  
We have really worked hard to follow standards, not go against them. 

Naming this table or referring to it as academic is very misleading and confusing. Because I call it academic, doesn't mean that it is. Because Eric calls it academic, doesn't mean that it is. I think that it is important to keep in mind that these are braille tables, and these braille tables allow us access to do academic work, that is evaluated in academic circles. This table is certainly not that. Neither is the one I created.  

Again, it is my opinion that we work to follow the standards every where possible. It is better for braille, it is better for the users, and it is better for Lib Louis.  

I hope this helps.

EricJHarvey commented:
Oh, ok. So it was deliberate. I thought it might be, since it is so different from the system I am familiar with. I'm happy with two tables, as long as the one that represents the documented Duxbury system is among them.
I don't know of any movement to update or change the Duxbury table. If David and Matt were working on that, it was between the two of them. It was not shared with the rest of the committee and has not been implemented in the Duxbury software or online documentation. At this point, then, I think it should be considered either an unpublished standard or personalized table. Either one is fine, but I don't think it can unilaterally replace the existing system.
I was invited to participate in that committee because at the time I was new to braille but had a lot of experience with Sumerian, Akkadian, and other ancient Near Eastern languages.
At the time, it was decided that the diacritics should resemble IPA braille more than UEB, and for good reason. UEB diacritics are bulky, which is fine in English, but in transliteration diacritics are far more common and having them more compact makes a better reading experience.
Again, I'm not advocating for one table over another,

EricJHarvey commented:
Also, I'm not tied to the name "Academic," but I think we need to differentiate the two tables. I couldn't differentiate it regionally, since these languages are no longer spoken but are studied in many places. Perhaps UEB-style and IPA-style might be more appropriate.

yeatersink commented:
Eric, 
I appreciate your feedback, but  the tables at DBT are still not officially finalized. Further, DBT made the official change to UEB years ago, and David spoke to us about that transition. Literally everyone knows that this change was to take place. It was a huge leap for DBT and the entire world.

We talked at length about the UEB transition. This was a huge transition for DBT. If it was not updated in this table, it may have ben because David was in the hospital for over a year during Covid, we literally were working with David in 22/23 on this project before he passed. 
Admittedly, we began with IPA because that was what DBT used before the transition took place. The  Tables were never set in stone, they couldn't have been, because we were working on them when he passed; very unexpectedly to us.

To make the transition to UEB is a natural transition; especially since it is being used literally everywhere. If you are making the claim that DBT has not updated the Biblical Languages table to reflect UEB, then I am going to check and verify that.  The conversation was that DBT was making that transition, and we were going to align  with those standards so that it would be natural  braille, and not something outlier that would make the Biblical Languages table difficult for new users. Even Here in Israel, they are updating the English to be UEB to align  with the rest of the world. 

If you are claiming that this table represents IPA for diacritics, then maybe calling it that would be a plausible solution. But you should give UEB a try, it works great with Akkadian- And it is nicer, because it is consistent.   

I am not trying to make a new braille, I am trying to stick with the rules and expectations of braille and our braille authority. It is my understanding that DBT has made that transition, and it only makes sense that we make that transition as well. 

It is nice to see you working on accessibility. This is really cool.  

By the way, if you want a bit of help, you can use Niv Louie. It makes building tables and Add Ons for NVDA a breeze.   check it out at nivlouie.com  It is the software we built and tried to get you and  Sarah  to be a part of.  It is finished now, and all you have to do is build a spread sheet and Niv Louie will do the rest, perfectly, and extremely fast.  

We are actually getting ready to present our papers at several conferences around the world this year because Niv Louie that we developed here in my PhD program literally has changed accessibility for ever.  I hope you give it a shot. 

@bertfrees  @egli  If you want me to look at this table closely, i am going to need a bit. This is a very busy time for me. If Eric needs this, than I don't want to stand in his way. 


You are very knowledgeable brother.

bertfrees commented:
Thanks both. My main goal was indeed to get rid of the term "Academic", because I find it too ambiguous, and it basically voids the non-"Academic" table (because why would a user want to pick that one?). If we are to include both tables, which I think is my preference at the moment, it needs to be clear from the names how the tables differ. "UEB-style vs. IPA-style" is a good direction.

EricJHarvey commented:
Matt, the Duxbury Biblical languages table does use UEB. The Semitic transliteration table does not.
Regional *English* braille codes have transitioned to UEB, but transliteration is not English, so that’s not necessarily relevant. Just like Spanish, French, and German all have different methods for representing diacritics that are not changing due to the adoption of UEB for English.
It’s not wrong to use UEB-style diacritics for transliteration, but it is a very different approach from the one we worked out back in 2017–2018.

I’m happy with two tables, and “UEB-style” and “IPA-style” seem like a perfectly good way to differentiate the two to me. At least, I can’t think of anything clearer.
@bertfrees would you like me to change the metadata and filenames or is that more easily accomplished on your end?

bertfrees commented:
@EricJHarvey Thanks for proposing to update the metadata and file names. Yes, you may. But don't worry too much about the correctness of all the fields (starting with `#-` and `#+`). I can still tweak where needed. Just make sure that there are a few lines explaining how this table differs from the other one. This will be useful in case what we come up with for the display name needs some more elaboration.

The display name should be concise, yet not compromise too much on conveying of information. The goal of the index name is to be as short as possible. To get an idea, check the names of all other tables in the file extra/generate-display-names/display-names.

yeatersink commented:
Greetings Eric, 

For the record, here is a link that speaks to the fact that UEB was intended to be the foundation for the Biblical Languages/ Semitic Scholars language package. 
https://www.duxburysystems.com/documentation/dbt$12.3/language_translation_tables/biblical_original_language_studies.htm

It also points out that this project began with me wanting to create access to tools in the academic arena.  We have been working on it since 2013. Also, I will place a copy of an email from David Holladay at the bottom of this post. This email shows  that Sarah and Ray began by sending emails about Greek and Hebrew, and David and I producing the table for an Academic setting began wen David and I began working on it in 2013/14.  David in this email says that this table was built on the Unified English, and continues to be a developing project. This email was written in October of 23. What this means is that the project was not complete. Why? We were continuing to work on it! And, if the transliteration table had not been updated, then perhaps it just not been done yet. Davids work was always free for us, so I never really pushed him beyond getting it started to begin with. 

@EricJHarvey  you speak about the code that we worked out in 2017/2018. By we are you speaking about Inbal Baum and I? Because that is who actually wrote the code in class at Bar Ilan University. The code Inbal Baum and I wrote together, and was introduced to DBT is the same code that is in the table in DBT today.   I have not looked, but I hope that your meta data actually gives credit where the credit is due. 

To be fair to you, you came to the committee in 2017 and your help is indeed valuable.

It is important to note that even the Semitic table page acknowledges that the transliteration table was built upon the Biblical language table, and claims that this continues to be a work in progress, and invites others to participate. 

The point is that the semitic table was built on the Biblical language table as the following link suggests. 
https://www.duxburysystems.com/documentation/dbt12.6/languages/semitic_tbl.htm

However, the Biblical languages table was built on the UEB as I argued above. The intention was never to stay in the old way of braille, the intention was to always to align  with UEB standards so we wouldn’t have several types of braille. Even the letters in the Transliteration table  worked to maintain consistency. The letters intended to represent semitic sounds take the form in braille as the Hebrew  letters. It does that  so we could have consistency and the braille would be natural.  

@egli  @bertfrees  2 tables are fine with me. I would ask that the metadata include credits for the entire committee and Inbal Baum and I for originally developing it.  

At the beginning  of building this project on DBT and  making it main stream, I was the one who took Ray's braille code for cantillation  and presented it to DBT, and that was when the official committee was formed.  Sarah was invited because of her experience with Greek and  Eric was invited later because of his experience with Akkadian. The only two of us who actually innovated and developed braille code was Ray and I.  Ray had a team of sighted scholars who helped him, and I had a team of scholars who helped me. The scholars that helped me spanned over 3 universities where I have studied. The scholars are literally from all over the world.  Together this committee worked to create unprecedented access  in this area of study. It has been my practice to include everyone in the credits. I realize not everyone who has published tables   here on Lib Louis has given the credit to those of us who have worked on this project. But i would ask that everyone be included in the credits. 

Here is a copy of the email. If anyone wants to read it, I am happy to forward it to them. This was davids response to an email sent to the entire committee: 

On Oct 24, 2023, at 4:15 PM, David Holladay <david@duxsys.com> wrote:


Sarah:
Unfortunately, most of my email has been wiped out when my computer crashed. So I really can offer much in the way of History.
Your history of this project starts with exchanging emails with Ray about Hebrew and Greek starting around 2010. My history starts with the phone call from Matt pleading the case for a new translator. What sealed the deal was his sending a sample file showing the critical apparatus and I realized no translator we had could you with that material.
As best as I can determine that phone call came in the spring of 2014. In November 2014 I formally made a copy of the English unified translator and named it as the foundation of the biblical translator. In July of 2015 Duxbury released a new version, I think 11.3, with a new biblical translator, which has been under continuous revision since then. The other significant date was July 2016 when the NFB gave the award for the project.
Perhaps Matt can remind me of the exact date or exact month when he made that phone call they got the whole experience started.
Collecting history is always good. It's a good way to be reminded of loose ends of the project that are undone. And I have to hope that everybody is safe and sound.
- David

yeatersink commented:
@EricJHarvey  @bertfrees  @egli  Here is another link that shows the intention of the Roman alphabet  to be represented by the UEB. This page shows the Semitic  languages table bbeing a sister table. 
I think that because the project was a work in process, things may have gotten a bit confusing. Not everything had gotten updated. In fact, I see where David started    citing the work that others had contributed, but the list of names never got added to the list. 

[cuneiform table duxbury systems](https://www.duxburysystems.org/dbtwebhelp/content/languages/biblical_original_languages_tbl.htm)

EricJHarvey commented:
@Bertfrees @yeatersink As I revisit our reasons for making a transliteration code that was different from (contracted) UEB, I’m realizing the base reason was compactness. I think “Compact” would be a better name than “IPA-style.” Would that work for you, to have Cuneiform (UEB) and Cuneiform (Compact)? I provide more reasoning and background  below:
There were two translation profiles implemented in Duxbury. One was a Biblical Original Languages table that accommodated the native scripts for Hebrew, Greek, Coptic, Ethiopic, and Syriac, as well as English text. Mat linked this above. This was developed by Matt, Ray, and Sarah in collaboration with David before I joined the committee, although I helped a bit in its later refinement. This translation profile allowed the translation of multilingual documents with text in several scripts.
I joined the committee to work on a separate translation profile, Semitic Transliteration. This is the one I link in the documentation to the table. Here the logic was different, as the appended email shows. Whereas the Biblical Original Languages table was meant to be inherently multi lingual and multi-script, this one was optimized to reading long stretches of transliterated languages, where English text would be  infrequent or not present at all. We opted to use single-cell patterns for diacritic marks instead of the two or three-cell UEB patterns, but the result was that the diacritic patterns would have interfered with the contraction indicators for contracted UEB. For example, dot 5 in this code indicates a dot below the following letter and dots 56 indicate a circumflex accent. While base characters do mostly align with uncontracted UEB, then, the diacritics do not and they create conflicts with them.
I also realize that, while we took some cues from the IPA code, our transliteration code differs enough from it that calling it “IPA-style” would be misleading.
To summarize, both tables align more-or-less with uncontracted UEB. Matt’s table also uses UEB diacritics. Mine uses bespoke diacritics inspired by some IPA symbols. Neither table supports contracted UEB.
So, all that to ask, would “Cuneiform (UEB)” and “Cuneiform (Compact)” be acceptable index names? 

Email from 2018:
From: David <david@duxsys.com>
Date: Fri, Jan 12, 2018 at 8:14 AM
Subject: Duxbury Work on Biblical and Semitic Languages

To all:

(Pay attention, there are a few questions to vote on in here.)

Ray raised some interesting issues with me in a private e-mail. I will
answer in an e-mail to the group.

This whole project started with a phone call I got on February 11, 2014
from Matthew Yeater. So we are close to the fourth anniversary of this
great endeavor.

Each Duxbury DBT program now has a "Biblical original languages studies". A
few months ago, we started on this new project, which I am calling
"Semitic Language Transcription".

During this process, we are making corrections to the "Biblical languages"
table as well: fixing many Greek vowel sequences, and adding Ethiopic.
Caryn has been helping with a number of adjustments to produce better
transitions from non-Roman script to Roman script when there are uppercase
and/or emphasis

The Biblical languages translator uses UEB contracted English, and can
handle a variety of scripts: Hebrew, Greek, Syriac, Coptic, Ethiopic,
Russian, and Arabic. We put in a critial analysis mode so that these
sequences could be read without excessive language switching indicators.

----------------

Since this Akkadian/Summarian translator works from complex sequences of
Roman script, we moved away from contracted UEB as our base.
Instead, we are basing our work on uncontracted UEB. This does give us a
lot of room to re-use braille that otherwise would be used for contractions.

I was more interested in tinkering before. I would like tinkering in
braille for the basic symbols used in Akkadian to be ended very soon.

I want people to vote on:
what to use for the half-ring (2 cells or 1 cell)
and what to use for the underdot. I actually prefer Ray's suggestion of the
dots 456 myself, but as a non-scholar, I am not eligible for this vote.

Yes, we are moving towards developing a code that will work with the
broadest group of Semitic language transcriptions as we can.
No, we will not force weird variations (some professor in Yale who has
their own system, etc.) to alter our work.

I will admit to toying with the idea of trying to render Brill
transliteration for Scholarly Hebrew as Hebrew braille. Sarah blew that
idea up in a few minutes.

-- David H

yeatersink commented:
Yes sir, I told you it was based on the UEB. And you are correct and that this does minimize the amount of Braille noise. The bigger problem is that we are stepping out of UEB that is the standard. I agree with you that calling it uncontracted is a bit misleading. I honestly think that having another table is going to be more confusing for the general reader that is reading regular UEP everywhere else. But if you need it then it's important that you have it.  When we first started making this table I don't think that any of us that were actually using Grill really understood the impact that the change to UEB was going to have and just exactly what the differences were. But now, the Nfb is making Braille for emojis, and they're trying to add it to the UEB table. UEB is literally the standard. And we can't call it un contracted because it's not really there and we can't call it contracted because it's not really that… Lol  This is why I understood to be updating the table to be full in alignment with UEB. It was the only thing that made sense.   If David never got the table updated in DBT, I don't know if I really want to change it. I absolutely love and adore him. Even that email that you shared shows that whatever I needed he worked on it and I wanna honor him. I just want to make sure that he gets credit, you can call  the table whatever  you want to brother. If he was still alive today I'm pretty sure we would've had it all updated by now. But if you want to keep it and you need it, then keep it. If you wanna call it compact you can do that too. Matt Yeater- Yeshurun On Feb 27, 2025, at 10:11 PM, Eric Harvey ***@***.***> wrote:﻿
  @bertfrees @yeatersink As I revisit our reasons for making a transliteration code that was different from (contracted) UEB, I’m realizing the base reason was compactness. I think “Compact” would be a better name than “IPA-style.” Would that work for you, to have Cuneiform (UEB) and Cuneiform (Compact)? I provide more reasoning and background  below:
There were two translation profiles implemented in Duxbury. One was a Biblical Original Languages table that accommodated the native scripts for Hebrew, Greek, Coptic, Ethiopic, and Syriac, as well as English text. Mat linked this above. This was developed by Matt, Ray, and Sarah in collaboration with David before I joined the committee, although I helped a bit in its later refinement. This translation profile allowed the translation of multilingual documents with text in several scripts.
I joined the committee to work on a separate translation profile, Semitic Transliteration. This is the one I link in the documentation to the table. Here the logic was different, as the appended email shows. Whereas the Biblical Original Languages table was meant to be inherently multi lingual and multi-script, this one was optimized to reading long stretches of transliterated languages, where English text would be  infrequent or not present at all. We opted to use single-cell patterns for diacritic marks instead of the two or three-cell UEB patterns, but the result was that the diacritic patterns would have interfered with the contraction indicators for contracted UEB. For example, dot 5 in this code indicates a dot below the following letter and dots 56 indicate a circumflex accent. While base characters do mostly align with uncontracted UEB, then, the diacritics do not and they create conflicts with them.
I also realize that, while we took some cues from the IPA code, our transliteration code differs enough from it that calling it “IPA-style” would be misleading.
To summarize, both tables align more-or-less with uncontracted UEB. Matt’s table also uses UEB diacritics. Mine uses bespoke diacritics inspired by some IPA symbols. Neither table supports contracted UEB.
So, all that to ask, would “Cuneiform (UEB)” and “Cuneiform (Compact)” be acceptable index names?
Email from 2018:
***@***.***
Date: Fri, Jan 12, 2018 at 8:14 AM
Subject: Duxbury Work on Biblical and Semitic Languages
To all:
(Pay attention, there are a few questions to vote on in here.)
Ray raised some interesting issues with me in a private e-mail. I will
answer in an e-mail to the group.
This whole project started with a phone call I got on February 11, 2014
from Matthew Yeater. So we are close to the fourth anniversary of this
great endeavor.
Each Duxbury DBT program now has a "Biblical original languages studies". A
few months ago, we started on this new project, which I am calling
"Semitic Language Transcription".
During this process, we are making corrections to the "Biblical languages"
table as well: fixing many Greek vowel sequences, and adding Ethiopic.
Caryn has been helping with a number of adjustments to produce better
transitions from non-Roman script to Roman script when there are uppercase
and/or emphasis
The Biblical languages translator uses UEB contracted English, and can
handle a variety of scripts: Hebrew, Greek, Syriac, Coptic, Ethiopic,
Russian, and Arabic. We put in a critial analysis mode so that these
sequences could be read without excessive language switching indicators.

Since this Akkadian/Summarian translator works from complex sequences of
Roman script, we moved away from contracted UEB as our base.
Instead, we are basing our work on uncontracted UEB. This does give us a
lot of room to re-use braille that otherwise would be used for contractions.
I was more interested in tinkering before. I would like tinkering in
braille for the basic symbols used in Akkadian to be ended very soon.
I want people to vote on:
what to use for the half-ring (2 cells or 1 cell)
and what to use for the underdot. I actually prefer Ray's suggestion of the
dots 456 myself, but as a non-scholar, I am not eligible for this vote.
Yes, we are moving towards developing a code that will work with the
broadest group of Semitic language transcriptions as we can.
No, we will not force weird variations (some professor in Yale who has
their own system, etc.) to alter our work.
I will admit to toying with the idea of trying to render Brill
transliteration for Scholarly Hebrew as Hebrew braille. Sarah blew that
idea up in a few minutes.
-- David H—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>



EricJHarvey left a comment (liblouis/liblouis#1728)
@bertfrees @yeatersink As I revisit our reasons for making a transliteration code that was different from (contracted) UEB, I’m realizing the base reason was compactness. I think “Compact” would be a better name than “IPA-style.” Would that work for you, to have Cuneiform (UEB) and Cuneiform (Compact)? I provide more reasoning and background  below:
There were two translation profiles implemented in Duxbury. One was a Biblical Original Languages table that accommodated the native scripts for Hebrew, Greek, Coptic, Ethiopic, and Syriac, as well as English text. Mat linked this above. This was developed by Matt, Ray, and Sarah in collaboration with David before I joined the committee, although I helped a bit in its later refinement. This translation profile allowed the translation of multilingual documents with text in several scripts.
I joined the committee to work on a separate translation profile, Semitic Transliteration. This is the one I link in the documentation to the table. Here the logic was different, as the appended email shows. Whereas the Biblical Original Languages table was meant to be inherently multi lingual and multi-script, this one was optimized to reading long stretches of transliterated languages, where English text would be  infrequent or not present at all. We opted to use single-cell patterns for diacritic marks instead of the two or three-cell UEB patterns, but the result was that the diacritic patterns would have interfered with the contraction indicators for contracted UEB. For example, dot 5 in this code indicates a dot below the following letter and dots 56 indicate a circumflex accent. While base characters do mostly align with uncontracted UEB, then, the diacritics do not and they create conflicts with them.
I also realize that, while we took some cues from the IPA code, our transliteration code differs enough from it that calling it “IPA-style” would be misleading.
To summarize, both tables align more-or-less with uncontracted UEB. Matt’s table also uses UEB diacritics. Mine uses bespoke diacritics inspired by some IPA symbols. Neither table supports contracted UEB.
So, all that to ask, would “Cuneiform (UEB)” and “Cuneiform (Compact)” be acceptable index names?
Email from 2018:
***@***.***
Date: Fri, Jan 12, 2018 at 8:14 AM
Subject: Duxbury Work on Biblical and Semitic Languages
To all:
(Pay attention, there are a few questions to vote on in here.)
Ray raised some interesting issues with me in a private e-mail. I will
answer in an e-mail to the group.
This whole project started with a phone call I got on February 11, 2014
from Matthew Yeater. So we are close to the fourth anniversary of this
great endeavor.
Each Duxbury DBT program now has a "Biblical original languages studies". A
few months ago, we started on this new project, which I am calling
"Semitic Language Transcription".
During this process, we are making corrections to the "Biblical languages"
table as well: fixing many Greek vowel sequences, and adding Ethiopic.
Caryn has been helping with a number of adjustments to produce better
transitions from non-Roman script to Roman script when there are uppercase
and/or emphasis
The Biblical languages translator uses UEB contracted English, and can
handle a variety of scripts: Hebrew, Greek, Syriac, Coptic, Ethiopic,
Russian, and Arabic. We put in a critial analysis mode so that these
sequences could be read without excessive language switching indicators.

Since this Akkadian/Summarian translator works from complex sequences of
Roman script, we moved away from contracted UEB as our base.
Instead, we are basing our work on uncontracted UEB. This does give us a
lot of room to re-use braille that otherwise would be used for contractions.
I was more interested in tinkering before. I would like tinkering in
braille for the basic symbols used in Akkadian to be ended very soon.
I want people to vote on:
what to use for the half-ring (2 cells or 1 cell)
and what to use for the underdot. I actually prefer Ray's suggestion of the
dots 456 myself, but as a non-scholar, I am not eligible for this vote.
Yes, we are moving towards developing a code that will work with the
broadest group of Semitic language transcriptions as we can.
No, we will not force weird variations (some professor in Yale who has
their own system, etc.) to alter our work.
I will admit to toying with the idea of trying to render Brill
transliteration for Scholarly Hebrew as Hebrew braille. Sarah blew that
idea up in a few minutes.
-- David H

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

EricJHarvey commented:
@Bertfrees I apologize, but for some reason every time I try to edit the files I get errors and it crashes. I think you may have to make the changes.
Here are my suggestions for edits, provided you are ok with “Compact” as a designation. Feel free to tweak as necessary.

For the translation table:
Filename: cuneiform-transliterated-compact.utb

First four lines of the file
# Cuneiform Transliteration and Normalization with Compact Diacritics
#
#-index-name: Cuneiform, Transliterated, Compact 
#-display-name: Transliterated Cuneiform with Compact Diacritics

And new text to be added before line 48 (that is, before the line that says “# Currently, it supports text in Sumerian, Akkadian (i.e., Babylonian”(
# This table uses compact definitions for diacritic marks developed
# specifically for transliteration by the team of scholars below.
# For a transliteration table that uses diacritic marks as defined
#  in Unified English Braille, see cuneiform-transliterated.utb.

For the YAML test:
Filename: cuneiform-transliterated-compact.yaml

And just change the target table, of course.

Thank you for the help!

bertfrees commented:
Thanks @EricJHarvey. I just did some final tweaks, then merge the test files, and finally rebased the branch onto the latest master so it is ready to be merged now.

EricJHarvey commented:
Thank you, @bertfrees, this all looks very good!
